### PR TITLE
[master] Avoid use of deprecated net-tools and add support for ip neighbour

### LINKED ISSUE #57541: [Tech-Debt] Avoid use of deprecated net-tools
**Is your feature request related to a problem? Please describe.**
The linux `net-tools` package has been deprecated since 2011, and provides commands including `arp`, `ifconfig`, `iptunnel`, `iwconfig`, `nameif`, `netstat`, and `route`.

Some salt modules make use of these commands, causing failures when they're not available (#57513).

**Describe the solution you'd like**
Code should be rewritten, preferring first native python (e.g. `pyroute2`) then the `iproute2` tools (e.g. `ip`), then only falling back to `net-tools` if that's all that is available. If every supported platform has already migrated to `iproute2` then the fallback is not required.

**Describe alternatives you've considered**
Continue to depend on systems providing the `net-tools` package. This won't work forever.

**Additional context**
Replacement commands:

net-tools | iproute2
-- | --
arp | ip neighbor
ifconfig | ip addr, ip link, ip -stats
iptunnel | ip tunnel
iwconfig | iw
nameif | ip link
netstat | ss
route | ip route

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Avoids use of deprecated net-tools and add support for ip neighbour with IPv4 ip_neighs and IPv6 ip_neighs6, similar to ip_addrs and ip_addrs6

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/57541

### New Behavior
Added support for ip neighbour with IPv4 ip_neighs and IPv6 ip_neighs6, and ensured that deprecated net-tools are no longer used when possible.

Also ported unittests to pytests and added tests for ip_neighs and ip_neighs6.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/network.py [hunk 1]
```diff
@@ -1333,6 +1333,60 @@ def ip_addrs6(interface=None, include_loopback=False, cidr=None):
...
+        comps = line.split()
+        if len(comps) < 5:
+            continue
+        if "." in comps[0]:
+            ret[comps[4]] = comps[0]
+
+    return ret
+
+
+ipneighs = salt.utils.functools.alias_function(ip_neighs, "ipneighs")
```
Comment (s0undt3ch) [inline salt/modules/network.py:1360]:
```suggestion
```
If this function is new, why support 2 names?

I'm guessing the existing aliases were to be backwards compatible with something?

File: salt/modules/vagrant.py [hunk 2]
```diff
@@ -627,9 +637,53 @@ def get_ssh_config(name, network_mask="", get_private_key=False):
...
+                "-oUserKnownHostsFile={UserKnownHostsFile} "
+                "-oControlPath=none "
+                "{User}@{HostName} ifconfig".format(**ssh_config)
+            )
+
+            log.info(
+                "Trying ssh -p %(Port)s %(User)s@%(HostName)s ifconfig", ssh_config
+            )
+            reply = __salt__["cmd.shell"](command)
+            log.info("--->\n%s", reply)
```
Comment (s0undt3ch) [inline salt/modules/vagrant.py:657]:
```suggestion
            log.info(
                "Trying ssh -p %(Port)s %(User)s@%(HostName)s ifconfig", ssh_config
            )
            reply = __salt__["cmd.shell"](command)
            log.info("ifconfig returned:\n%s", reply)
```

File: tests/pytests/unit/modules/test_network.py [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
 def configure_loader_modules():
-    return {networkmod: {}}
+    ## return {networkmod: {}}
+    opts = salt.config.DEFAULT_MINION_OPTS.copy()
+    utils = salt.loader.utils(
+        opts, whitelist=["network", "path", "platform", "stringutils"]
+    )
+    return {
+        networkmod: {"__utils__": utils},
+    }
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_network.py:28]:
```suggestion
    ## return {networkmod: {}}
    utils = salt.loader.utils(
        minion_opts, whitelist=["network", "path", "platform", "stringutils"]
    )
    return {
        networkmod: {"__utils__": utils},
    }
```
Add `minion_opts` to `configure_loader_modules()` as an argument

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/modules/network.py (hunk 1 above)
Comment (dmurphy18) [reply salt/modules/network.py:1360]: Not being backwards compatible with anything but maintaining existing convention, as stated in the PR, similar to ip_addrs and ip_addrs6, if we split it up for ```ip addr```, keeping it similar for ```ip neighbour``` , noting ```man 8 ip-neighbour``` shows 'ip neigh'

Comment (s0undt3ch) [reply salt/modules/network.py:1360]: ok

**Review by s0undt3ch — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
@dmurphy18 this has failing tests related to your changes that need to be resolved.

**Comment by dmurphy18** [comment]
@Ch3LL The original PR had all tests passing, after the suggested additions, failures :) working the fixes now

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**